http/vibe/http/server.d

```diff
@@ -1141,10 +1141,18 @@ final class HTTPServerRequest : HTTPRequest {
 	*/
 	@property string rootDir()
 	const @safe {
-		import std.range.primitives : walkLength;
-		auto depth = requestPath.bySegment.walkLength;
+		import std.algorithm.searching : count;
+		auto depth = requestPath.bySegment.count!(s => s.name.length > 0);
+		if (!requestPath.endsWithSlash) depth--;
 		return depth == 0 ? "./" : replicate("../", depth);
```

Comment:
Hmm that looks potentially dangerous. `count` returns an unsigned number.
Consider:

```d
/+ dub.sdl:
dependency "vibe-d:core" version="~master"
+/
import vibe.core.path;
import std.stdio;
import std.algorithm;

void main()
{
	auto p = InetPath("");
	p.bySegment.writeln;
	p.endsWithSlash.writeln;
	size_t depth = p.bySegment.count;
	if (!p.endsWithSlash) depth--;
	depth.writeln;
}
```

https://run.dlang.io/is/toQxsC

Reply:
Good catch! Fixed and rebased.